runtime/src/test/scala/akka/grpc/GrpcClientSettingsSpec.scala

```diff
@@ -169,6 +171,32 @@ class GrpcClientSettingsSpec extends WordSpec with Matchers with ScalaFutures {
       intercept[IllegalArgumentException] {
         GrpcClientSettings.fromConfig("project.WithNoServiceName")
       }.getMessage should be("requirement failed: Configuration must contain a service-name")
+
+    }
+    "fail fast when no service discovery is configured on the actor system" in {
+      intercept[IllegalArgumentException] {
+        val settings = GrpcClientSettings.usingServiceDiscovery("a-downstream-service")
+      }
+    }
+
+    "use the service discovery configured on the actor system" in {
+      // given an ActorSystem with a ServiceDiscovery method configured
+      val sdConfig =
+        s"""
+          |akka.discovery {
+          |  method = "${classOf[FakeServiceDiscovery].getName}"
+          |}
+        """.stripMargin
+      val actorSystem = ActorSystem("test-with-service-discovery", ConfigFactory.parseString(sdConfig))
+
+      // invoking usingServiceDiscovery provides a GrpcClientSettings instance
+      // that uses the ServiceDiscovery in the ActorSystem
+      val settings = GrpcClientSettings.usingServiceDiscovery("a-downstream-service")(actorSystem)
+      settings.serviceDiscovery should not be null
     }
+
   }
 }
+class FakeServiceDiscovery extends SimpleServiceDiscovery{
+  override def lookup(lookup: Lookup, resolveTimeout: FiniteDuration): Future[Resolved] = ???
```

Comment:
How does this result in a non-null service discovery?

Reply:
```scala
      // given an ActorSystem with a ServiceDiscovery method configured
      val sdConfig =
        s"""
          |akka.discovery {
          |  method = "${classOf[FakeServiceDiscovery].getName}"
          |}
        """.stripMargin
      val actorSystem = ActorSystem("test-with-service-discovery", ConfigFactory.parseString(sdConfig))
```
this code makes sure the `ActorSystem` will have an instance of `FakeServiceDiscovery` as a service discovery mechanism. So, creating a `GrpcClientSettings` instance via `GrpcClientSettings.usingServiceDiscovery` will use that service discovery instance. But, if a user tried an operation implying a `lookup` with that service discovery, then the operation would fail.

This test and the previous one assert that if there's a service discovery it's used and if there's none things fail fast.